typing/typecore.ml

```diff
@@ -3865,16 +3872,27 @@ and type_expect_
         exp_attributes = sexp.pexp_attributes;
         exp_env = env }
   | Pexp_array(sargl) ->
-      let ty = newgenvar() in
-      let to_unify = Predef.type_array ty in
-      with_explanation (fun () ->
-        unify_exp_types loc env to_unify (generic_instance ty_expected));
+      let ty_elt, exp_type =
+        let ty_expected = generic_instance ty_expected in
+        if is_principal ty_expected &&
+           match get_desc (expand_head env ty_expected) with
+           | Tconstr (path, [], _) -> Path.same path Predef.path_floatarray
+           | _ -> false
+        then
+          Predef.type_float, Predef.type_floatarray
```

Comment:
Same here as above: no need to return this here. The `ty_expected` should work fine.

Reply:
Fixed as well, thanks.